back/dora/core/views.py

```diff
@@ -7,30 +7,43 @@
 from django.utils.text import get_valid_filename
 from rest_framework import permissions
 from rest_framework.decorators import api_view, parser_classes, permission_classes
-from rest_framework.exceptions import ValidationError
+from rest_framework.exceptions import PermissionDenied, ValidationError
 from rest_framework.parsers import FileUploadParser
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.request import Request
 from rest_framework.response import Response
 
 from dora.services.models import Service
-from dora.structures.models import Structure
+from dora.structures.models import Structure, StructureMember
 
 logger = logging.getLogger(__name__)
 
 
-def _validate_upload(file_obj):
-    if file_obj.size > settings.MAX_UPLOAD_SIZE_MB * 1024 * 1024:
+def _validate_upload(file_name: str, file_size: int) -> None:
+    if file_size > settings.MAX_UPLOAD_SIZE_MB * 1024 * 1024:
         raise ValidationError("FILE_TOO_BIG")
-    if file_obj.name.split(".")[-1] not in settings.ALLOWED_UPLOADED_FILES_EXTENSIONS:
+    if file_name.split(".")[-1] not in settings.ALLOWED_UPLOADED_FILES_EXTENSIONS:
```

Comment:
Le fichier uploadé n'aura pas forcément d'extension (pas de `.`).
```suggestion
    if "." not in file_name or file_name.split(".")[-1] not in settings.ALLOWED_UPLOADED_FILES_EXTENSIONS:
```